Changed support from deprecated private apps to custom apps

GameSpecU opened the pull request.

dawid-focht committed:
Updated usage for custom apps instead of deprecated private apps

dawid-focht committed:
Updated Workflow

dawid-focht committed:
rebuild

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
fix defaults

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

dawid-focht committed:
deprecated rmSync updated to rmdirSync

dawid-focht committed:
Autolink fix

dawid-focht committed:
[Build] Autolink fix

dawid-focht committed:
Simplified PR comment

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
debug

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

dawid-focht committed:
Changed: theme name from branch name instead of PR id

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

GameSpecU committed:
Cross-stores themes support

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

matthew-petrie commented on action.yml:
```diff
@@ -10,25 +10,25 @@ inputs:
+    default: "."
   SHOPIFY_ALLOW_LIVE_THEME_DEPLOYMENT:
     description: "If deploying to a 'live' Shopify theme this must be set to 'true'"
     required: false
+    default: "false"
```
Suspect required now needs to change to `false` as it has a default value

GameSpecU replied:
@matthew-petrie 
Could you extend your thought?
I belive, default value was needed here because of null exception

matthew-petrie commented on src/index.help.ts:
```diff
@@ -17,7 +16,21 @@ import {
+  `Shopify Theme Actions for :${SHOPIFY_AUTH.storeUrl}`;
+
+export const getThemeName = (): string => {
+  let branch: string;
+  if (process.env.GITHUB_EVENT_NAME === "pull_request") {
+    branch = process.env.GITHUB_HEAD_REF!;
+  } else {
+    branch = process.env.GITHUB_REF_NAME!;
+  }
```
```suggestion
  const branch = (process.env.GITHUB_EVENT_NAME === "pull_request") ? process.env.GITHUB_HEAD_REF! : process.env.GITHUB_REF_NAME!
```

matthew-petrie commented on src/index.help.ts:
```diff
@@ -17,7 +16,21 @@ import {
+    branch = process.env.GITHUB_REF_NAME!;
+  }
+  const branchParts = branch.split("/");
+  branch = branchParts.pop()!;
```
Safer to use a different variable name that is a constant

matthew-petrie commented on src/index.help.ts:
```diff
@@ -17,7 +16,21 @@ import {
+  }
+  const branchParts = branch.split("/");
+  branch = branchParts.pop()!;
+  return branch;
```
This should still include the PR number if it is from a PR i.e. `PR ${pullRequestNumber}`

matthew-petrie commented on src/shopify.ts:
```diff
@@ -70,6 +75,33 @@ export const deployTheme = async (
   });
 };
 
+export const duplicateLive = async (SHOPIFY_AUTH: shopifyAuth, id: number): Promise<void> => {
+  !fs.existsSync(`./.shopify-tmp/`) && fs.mkdirSync(`./.shopify-tmp/`, { recursive: true });
+  await themeKit.command(
+    "download",
+    {
+      password: SHOPIFY_AUTH.password,
+      store: SHOPIFY_AUTH.storeUrl,
+      live: true,
+      noIgnore: true,
+      dir: "./.shopify-tmp/",
+    },
+    { logLevel: "all" }
+  );
+  await themeKit.command(
+    "deploy",
+    {
+      password: SHOPIFY_AUTH.password,
+      store: SHOPIFY_AUTH.storeUrl,
+      themeId: id,
+      noIgnore: true,
+      dir: "./.shopify-tmp/",
+    },
+    { logLevel: "all" }
+  );
+  fs.rmdirSync("./.shopify-tmp/", { recursive: true });
+};
```
Great addition, something I've been wanting to add for a while! Can you add some documentation into the readme?

GameSpecU replied:
Yeah, i'll add in some free time soon ;)

matthew-petrie commented on src/shopify.ts:
```diff
@@ -70,6 +75,33 @@ export const deployTheme = async (
   });
 };
 
+export const duplicateLive = async (SHOPIFY_AUTH: shopifyAuth, id: number): Promise<void> => {
```
```suggestion
export const duplicateLiveTheme = async (SHOPIFY_AUTH: shopifyAuth, id: number): Promise<void> => {
```

matthew-petrie commented on src/shopify.ts:
```diff
@@ -92,6 +124,7 @@ export const createOrFindThemeWithName = async (
         `Shopify theme with name '${shopifyThemeName}' should have been created and the theme found in Shopify however the theme cannot be found in Shopify.`
       );
     }
+    await duplicateLive(SHOPIFY_AUTH, shopifyTheme.id);
```
can we add a comment as to why we need to do this

matthew-petrie commented on src/index.help.ts:
```diff
@@ -74,6 +90,11 @@ export const deployment = async (
 
-  const message = `:tada: Shopify theme has been deployed to theme id '${shopifyThemeId}' at '${SHOPIFY_AUTH.storeUrl}'. The theme can be previewed at: ${themePreviewUrl}`;
-  await createReplaceComment(message, UNIQUE_HIDDEN_COMMENT_STRING, shopifyThemeId, GITHUB_AUTH);
+  const message = `:tada: '${SHOPIFY_AUTH.storeUrl}' preview at: \n <${themePreviewUrl}>`;
```
There is benefit to keeping the theme id in the comment as a separate item, so I think this should be reverted to what it currently is

GameSpecU replied:
@matthew-petrie 
Just wanted to simplify the comment - but yeah, the ID could be as a separate item.
Anyway:
From my own exprience there should be also Store url, since in some cases, devs are working on different environments within the same theme:
E.g. The company, i am working for, uses different stores for Staging, Production, sometimes even more ;)

GameSpecU replied:
@matthew-petrie
 i've reviewed - actually there still exists ID as a separate item - in hidden comment, i can't see any benefits of keeping them in the visible part additionally since preview url contains also themeid
What do you think?

matthew-petrie commented on src/index.help.ts:
```diff
@@ -17,7 +16,21 @@ import {
+export const UNIQUE_HIDDEN_COMMENT_STRING = (SHOPIFY_AUTH: shopifyAuth): string =>
+  `Shopify Theme Actions for :${SHOPIFY_AUTH.storeUrl}`;
+
+export const getThemeName = (): string => {
```
Great addition of the branch name

matthew-petrie reviewed (changes requested):
Some great changes here - a couple of thoughts below

GameSpecU committed:
Apply suggestions from code review

Co-authored-by: Matthew Petrie <dev@matthewpetrie.co.uk>

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
small fixes

dawid-focht committed:
debug messages

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
debug messages

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

dawid-focht committed:
debug messages

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	src/shopify.ts

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Live Download error fixing
troubleshooting error: `requested theme was not found` when downloading using ``--live`

dawid-focht committed:
Merge remote-tracking branch 'origin/main'

github-actions[bot] committed:
Commit from GitHub Actions (pull-request)
